x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/TransportNodeDeprecationCheckAction.java

```diff
@@ -96,12 +101,28 @@ protected NodesDeprecationCheckResponse newResponse(
 
     NodesDeprecationCheckAction.NodeResponse nodeOperation(
         NodesDeprecationCheckAction.NodeRequest request,
-        List<BiFunction<Settings, PluginsAndModules, DeprecationIssue>> nodeSettingsChecks
+        List<
+            DeprecationChecks.NodeDeprecationCheck<
+                Settings,
+                PluginsAndModules,
+                ClusterState,
+                XPackLicenseState,
```

Comment:
We don't actually need the license state anywhere in 8.x (yet), but I figured while I was doing all this work to make 8.x more consistent with 7.x I might as well pull this in so that we don't have to come back and do it again later.